File: src/test/java/Utils/Config/BaseClass.java
package Utils.Config;


import Helpers.CommonMethods;
import Helpers.FetchProperties;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;

import java.time.Duration;
import java.util.HashMap;
import java.util.Objects;

public class BaseClass {
    public  static WebDriver driver;
    public static WebDriverWait wait;
    public static PageFactory pageFactory;

    @BeforeClass
    public void  initializeBrowser(){
        String downloadPath = CommonMethods.myPath("/src/test/java/DownloadPath");
        String path = "C:\\TotalConcepts\\SeleniumPractice\\src\\test\\java\\DownloadPath";


        WebDriverManager.chromedriver().setup();
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--start-maximized");
        driver = new ChromeDriver(options);
        HashMap<String ,Object> chromeDownload = new HashMap<>();
        chromeDownload.put("download.default_directory",path);
        options.setExperimentalOption("prefs",chromeDownload);
        wait = new WebDriverWait(driver, Duration.ofSeconds(50));
        pageFactory = new PageFactory();
        driver.get(FetchProperties.getEnvironmentVariablePath("BaseURl"));
    }
    @AfterClass
    public void logout(){
        driver.quit();
    }
}

File: cPractice/BaseClass.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata;
using System.Text;
using System.Threading.Tasks;

namespace cPractice
{
    public class BaseClass
    {
        public static IWebDriver Driver;
        public static WebDriverWait wait;

        [SetUp] 
        public void intialize()
        {
            ChromeOptions options = new ChromeOptions();
            options.AddArguments("Start-Maximized");
            Driver = new ChromeDriver(options);
            Driver.Url = TestContext.Parameters.Get("BaseUrl");
            wait = new WebDriverWait(Driver,TimeSpan.FromSeconds(90));


        }

        [TearDown]
        public static void BrowserClose()
        {
            Driver.Close();
            Driver.Dispose();

        }
    }
}
